pkg/agent/agent.go

```diff
@@ -288,7 +288,7 @@ func rerunAsRoot(workspaceInfo *provider2.AgentWorkspaceInfo, log log.Logger) (b
 	}
 
 	// call ourself
-	args := []string{binary}
+	args := []string{"--preserve-env", binary}
```

Comment:
This preserves all environment variables from the initial session into the `sudo` session. Technically, this could be reduced down to `--preserve-env=SSH_AUTH_SOCK` to limit only the SSH Agent socket being passed into the `sudo` session.

Reply:
Just to note, the preservation of the environment variables is necessary regardless of source OS. This is all about the target OS and how the Docker socket has been configured from an access standpoint. So any user using a target where Docker requires admin rights would have been impacted by this.